### ISSUE TITLE
site_frontpage = `user` menu link is not active

### ISSUE THREAD

**Issue by Gormartsen** [backdrop/backdrop-issues#2413]
Issue in router_item

For path `user` (and link with url `user`) router_item path is `user/%`

It happens due to next code:

https://github.com/backdrop/backdrop/blob/1.x/core/modules/user/user.pages.inc#L553
```
function user_page() {
  global $user;
  if ($user->uid) {
    menu_set_active_item('user/' . $user->uid);
    return menu_execute_active_handler(NULL, FALSE, 'menu_default_route_handler');
  }
  else {
    return backdrop_get_form('user_login');
  }
}
```

**Comment by Gormartsen** [comment]
PS: For authorized users.

**Comment by Gormartsen** [comment]
There is an extra case:
- menu link in primary navigation pointed to <front>
- site_frontpage set to empty (default) or user (it is default).
- primary navigation is a menu in Header Block

In this case, menu item will not get class active on front page.

**Comment by Gormartsen** [comment]
PR https://github.com/backdrop/backdrop/pull/1683 is POC patch.

**Comment by Gormartsen** [comment]
I am sorry if I used label not properly! `canadian way eh?`

**Comment by klonos** [comment]
Labels are just right @Gormartsen 👍

**Comment by Gormartsen** [comment]
Yes, if profile page will be removed entirely,  there is no need to do so.

Please check commit: https://github.com/backdrop/backdrop/commit/b3cf0681406832ea677d21beb438624a746b1622

It was commit, where module node was made optional and site_frontpage become `user` by default.

So if `user` will be removed, all default values for site_frontpage need to be updated as well.

**Comment by quicksketch** [comment]
I'm not comfortable with making exceptions in the menu system itself for the weird `/user` path, but I think we could avoid this problem entirely just by fixing the menu callback `user_page()` to restore the original menu path after rendering the profile page.

I filed a PR at https://github.com/backdrop/backdrop/pull/1727. @Gormartsen could you try that out?

**Comment by Gormartsen** [comment]
@quicksketch tested. See http://1727.backdrop.backdrop.qa.backdropcms.org

I changed QA sandbox to next settings:
- added home link to user menu
- set front_page (home) to `user`

Results:

### anonymous user:
 - `[passed]` user/login - active in menu when clicked 
 - `[not passed]` on home page with login form `user/login` link is not active. I marked it as not passed due to UX. It is strange to see login form and do not have login link being active.

### authorised user:
 - `[not passed]` after authorization user redirected to `user/1`. 
   - `user` (my account)  is not active.
   - `<front>` (home)  is not active.
- `[not passed]` on `user` (my account) page
  -  tabs are missing  due to  tab_root for view and edit  is `user/%`
  - `<front>` (home)  is not active.

**Comment by Gormartsen** [comment]
> I'm not comfortable with making exceptions

I understand it. Another way to fix it is completely rewrite user_menu by adding `user/edit` `user/view` items (for tabs on my account page) and remove `menu_execute_active_handler` from `user_page()`

**Comment by quicksketch** [comment]
> Results:

Well, rats. Not working quite yet then. Let's remove this from the bug fix release and keep working on suitable solutions.
